utils.h

```diff
@@ -0,0 +1,109 @@
+/**
+ * ----------------------------------------------------------------------------
+ * Company    : SLAC National Accelerator Laboratory
+ * ----------------------------------------------------------------------------
+ * Description: Misc utility code
+ * ----------------------------------------------------------------------------
+ * This file is part of 'cexp'. It is subject to the license terms in the
+ * LICENSE.txt file found in the top-level directory of this distribution,
+ * and at:
+ *    https://confluence.slac.stanford.edu/display/ppareg/LICENSE.html.
+ * No part of 'cexp', including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE.txt file.
+ * ----------------------------------------------------------------------------
+ **/
+#pragma once
+
+#include <stddef.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <string.h>
+
+#define MAX_MACRO_NAME_LEN 128
+
+/**
+ * Expands a string with macros in it.
+ * Macros must be surrouned by ${}, such as ${PWD} or ${TEST}. Escape
+ * sequences are also supported (i.e. \${PWD} will not be expanded).
+ * Macros in the format $TEST are not recognized or expanded.
+ * \param str Input string
+ * \param resolveMacro Function to return the value of a macro. Returns NULL
+ * if the macro doesn't exist.
+ * \returns heap allocated string containing the transformed input.
+ * The underlying buffer may be larger than the string contained within.
+ */
+static inline char*
+expand_macro_string(const char* str, const char*(*resolveMacro)(const char*))
```

Comment:
resolveMacro is not used by the function. Was the intention to use it in line 62?

Reply:
Yes